submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/ExperimentManager.java

```diff
@@ -85,13 +87,33 @@ private ExperimentManager(Submitter submitter) {
    */
   public Experiment createExperiment(ExperimentSpec spec) throws SubmarineRuntimeException {
     checkSpec(spec);
+
+    // Submarine sdk will get experimentID and JDBC URL from environment variables in each worker,
+    // and then log experiment metrics and parameters to submarine server
+    ExperimentId id = generateExperimentId();
+    String url = getSQLAlchemyURL();
+    spec.getMeta().getEnvVars().put(RestConstants.JOB_ID, id.toString());
+    spec.getMeta().getEnvVars().put(RestConstants.SUBMARINE_TRACKING_URI, url);
+
     Experiment experiment = submitter.createExperiment(spec);
-    experiment.setExperimentId(generateExperimentId());
+    experiment.setExperimentId(id);
+
+    spec.getMeta().getEnvVars().remove(RestConstants.JOB_ID);
```

Comment:
I don't quite understand why put the environment to the spec and then delete them?

Reply:
Those two environment variables are for internal usage, we set these environment variables in pods.
But we would not like to see them in the experiment spec.